File: MyApplication/app/src/main/java/com/cryptonites/homey/cfd/Appliance.java
package com.cryptonites.homey.cfd;

import org.json.JSONObject;

/**
 * This class will contain the features of an appliance and will give it to
 */

public class Appliance {

    private final String name;
    private int powerLevel;
    private boolean state;

    public Appliance(String name){
        this.name = name;
        powerLevel = 0;
        state = false;
    }

    public Appliance(String name, boolean state) {
        this.name = name;
        this.state = state;
        powerLevel = 0;
    }

    public Appliance(Appliance apl) {
        this.name = apl.getName();
        this.state = apl.getState();
        this.powerLevel = this.getPowerLevel();
    }

    public Appliance(String name, int powerLevel){
        this.name = name;
        this.powerLevel = powerLevel;
        state = false;
    }

    public Appliance(String name, int powerLevel, boolean state) {
        this.name = name;
        this.powerLevel = powerLevel;
        this.state = state;
    }

    public String getName() {
        return name;
    }

    public int getPowerLevel() {
        return powerLevel;
    }

    public void setPowerLevel(int powerLevel) {
        this.powerLevel = powerLevel;
    }

    public boolean getState() {
        return state;
    }

    public void setState(boolean state) {
        this.state = state;
    }

    public JSONObject getJsonObject(){
        JSONObject applJSON = new JSONObject();
        try{
            applJSON.put("applName", this.name);
            applJSON.put("state", (this.state)?"true":"false");
            applJSON.put("powerLevel", this.powerLevel);
        }catch (Exception e) {e.printStackTrace();}
        return applJSON;
    }
}

File: Pi_Interface/Appliance.py
class Appliance:
    def __init__(self, applName, state, powerLevel, pinNo):
        self.applName = applName
        self.state = state
        self.powerLevel = powerLevel
        self.pinNo = pinNo

    def __setstate__(self, state):
        self.state = state

    def __getstate__(self):
        return self.state

    def __setpowerLevel__(self, powerLevel):
        self.powerLevel = powerLevel

    def __getpowerLevel__(self):
        return self.powerLevel

    def __get__(self):
        return {"name": self.applName, "state": self.state, "powerLevel": self.powerLevel}

    def __getStringObject__(self):
        if self.state:
            strState = "true"
        else:
            strState = "false"
        return "{name: " + self.applName + ", state: " + strState + ", powerLevel: " + str(self.powerLevel) + "}\n"

    def startLed(self):
        GPIO.output(self.pinNo, True)

    def stopLed(self):
        GPIO.output(self.pinNo, False)
